packages/stream-node/writer.js

```diff
@@ -22,17 +22,11 @@ export const makeNodeWriter = writer => {
     X`Cannot convert Node.js object mode Writer to AsyncIterator<undefined, Uint8Array>`,
   );
 
-  const finished = new Promise(resolve => {
+  const finalIteration = new Promise((resolve, reject) => {
```

Comment:
Unfortunately I think that means we lose the rejection in case the stream finishes because of an error.

Reply:
We concluded out-of-band that a stream should terminate with either finish or error exclusively and that close may follow after that, but is not guaranteed. We agreed to keep this block of code mostly the same (I’ve appended a commit to document the choice).

Reply:
Yeah apologies, I initially missed the `writer.on('error', reject);` in my previous review.